x-pack/plugins/security_solution/public/resolver/view/panels/panel_content_related_counts.tsx

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo, useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiBasicTableColumn, EuiButtonEmpty, EuiSpacer, EuiInMemoryTable } from '@elastic/eui';
+import { FormattedMessage } from 'react-intl';
+import { CrumbInfo, StyledBreadcrumbs } from './panel_content_utilities';
+
+import * as event from '../../../../common/endpoint/models/event';
+import { ResolverEvent, ResolverNodeStats } from '../../../../common/endpoint/types';
+
+/**
+ * This view gives counts for all the related events of a process grouped by related event type.
+ * It should look something like:
+ *
+ * | Count                  | Event Type                 |
+ * | :--------------------- | :------------------------- |
+ * | 5                      | DNS                        |
+ * | 12                     | Registry                   |
+ * | 2                      | Network                    |
+ *
+ */
+export const EventCountsForProcess = memo(function EventCountsForProcess({
+  processEvent,
+  pushToQueryParams,
+  relatedStats,
+}: {
+  processEvent: ResolverEvent;
+  pushToQueryParams: (arg0: CrumbInfo) => unknown;
+  relatedStats: ResolverNodeStats;
+}) {
+  interface EventCountsTableView {
+    name: string;
+    count: number;
+  }
+
+  const relatedEventsState = { stats: relatedStats.events.byCategory };
+  const processName = processEvent && event.eventName(processEvent);
+  const processEntityId = event.entityId(processEvent);
+  const totalCount = Object.values(relatedStats.events.byCategory).reduce(

```

Comment:
can you add a code comment (or move it to a new models file) so that its clear that this determines what it determines.

Reply:
https://github.com/elastic/kibana/pull/69098/commits/6c20672a26cd8b93968b992fbdbc15a57b776b06